src/physics/utils/pio_reader.F90

```diff
@@ -0,0 +1,751 @@
+module pio_reader
+    use pio,            only : file_desc_t
+    use ccpp_io_reader, only : ccpp_io_reader_t
+    use shr_kind_mod,   only : cl=>shr_kind_cl
+    
+    implicit none
+    private
+  
+    public :: pio_reader_t
+
+    !Error code parameters
+    integer, parameter :: pio_inq_dim_id_err      = 1
+    integer, parameter :: pio_inq_dim_len_err     = 2
+    integer, parameter :: pio_inq_var_id_err      = 3
+    integer, parameter :: pio_inq_var_info_err    = 4
+    integer, parameter :: bad_var_rank_err        = 5
+    integer, parameter :: too_high_rank_err       = 6
+    integer, parameter :: pio_get_var_err         = 7
+    integer, parameter :: not_char_type_err       = 8
+
+    type :: sima_pio_fh_type
+        logical            :: is_file_open = .false.  !Is NetCDF file currently open?
+        type(file_desc_t)  :: pio_fh                  !PIO File handle type
+        character(len=cl)  :: file_path              !Local path to NetCDF file
+    end type
```

Comment:
Since this is a private class in this module, would a name like `file_handle_t` be better? I took me a minute to figure out what `fh` was, and I think the `sima` and `pio` will be pretty obvious from the context.

Reply:
Works for me, done!